cycler.py

```diff
@@ -124,6 +124,43 @@ def keys(self):
         """
         return set(self._keys)
 
+    def change_key(self, old, new):
+        """
+        Change a key in this cycler to a new name.
+        Modification is performed in-place.
+
+        Does nothing if the old key is the same as the new key.
+        Raises a ValueError if the new key is already a key.
+        Raises a KeyError if the old key isn't a key.
+
+        """
+        if old == new:
+            return
+        if new in self._keys:
+            raise ValueError("Can't replace %s with %s, %s is already a key" %
+                             (old, new, new))
+        if old not in self._keys:
+            raise KeyError("Can't replace %s with %s, %s is not a key" %
+                           (old, new, old))
+
+        self._keys.remove(old)
```

Comment:
and the original problem is solved, and the remaining exception turns out to be an error in the test I wrote. Now, I'll add a some more tests to make sure what we just encountered doesn't happen again.